Lib/test/test_threading.py

```diff
@@ -722,7 +730,80 @@ def func():
         _, out, err = assert_python_ok("-c", code)
         data = out.decode().replace('\r', '')
         self.assertEqual(err.decode('utf-8'), "")
-        self.assertEqual(data, "Thread-1 (func)\nTrue\nTrue\n")
+        self.assertEqual(data,
+                         "Thread-1 (func) Thread\n"
+                         "True\n"
+                         "True\n"
+                         "True\n")
+
+    @unittest.skipIf(sys.platform in platforms_to_skip, "due to known OS bug")
+    @support.requires_fork()
+    @unittest.skipUnless(hasattr(os, 'waitpid'), "test needs os.waitpid()")
+    def test_main_thread_after_fork_from_foreign_thread(self, create_dummy=False):
+        code = """if 1:
+            import os, threading, sys, traceback, _thread
+            from test import support
+
+            def func(lock):
+                ident = threading.get_ident()
+                if %s:
+                    # call current_thread() before fork to allocate DummyThread
+                    current = threading.current_thread()
+                    print(current.name, type(current).__name__)
+                print(ident in threading._active)
```

Comment:
You may add prefix for readability (need to update the expected output):

```suggestion
                print("ident in _active", ident in threading._active)
```